x-pack/plugins/files/public/components/image/components/blurhash.tsx

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { decode } from 'blurhash';
+import type { FunctionComponent } from 'react';
+import { useEuiTheme } from '@elastic/eui';
+import React, { useRef, useEffect } from 'react';
+import { css } from '@emotion/react';
+
+interface Props {
+  visible: boolean;
+  hash: string;
+  width: number;
+  height: number;
+  isContainerWidth: boolean;
+}
+
+export const Blurhash: FunctionComponent<Props> = ({
+  visible,
+  hash,
+  width,
+  height,
+  isContainerWidth,
+}) => {
+  const ref = useRef<null | HTMLCanvasElement>(null);
+  const { euiTheme } = useEuiTheme();
+  useEffect(() => {
+    try {
+      const pixels = decode(hash, width, height);
+      const ctx = ref.current!.getContext('2d')!;
+      const imageData = ctx.createImageData(width, height);
+      imageData.data.set(pixels);
+      ctx.putImageData(imageData, 0, 0);
+    } catch (e) {
+      // ignore if something goes wrong loading the blurhash
+    }
+  }, [hash, width, height]);
+  return (
+    <canvas
```

Comment:
note: as discussed, if BlurHash causes performance problems, we could limit the `<canvas>` element size, then convert `<canvas>` to URL object, and render it as `<img>` element:

```suggestion
    <img src={urlObject}
```

Reply:
This was actually causing issues at higher resolutions, I think we will have to go with a the image approach.